digiroad2-oracle/src/test/scala/fi/liikennevirasto/digiroad2/VerificationServiceSpec.scala

```diff
@@ -0,0 +1,114 @@
+package fi.liikennevirasto.digiroad2.util
+
+import com.jolbox.bonecp.{BoneCPConfig, BoneCPDataSource}
+import fi.liikennevirasto.digiroad2.{DigiroadEventBus, RoadLinkService, VerificationService}
+import fi.liikennevirasto.digiroad2.oracle.OracleDatabase
+import fi.liikennevirasto.digiroad2.verification.oracle.VerificationDao
+import org.mockito.Mockito._
+import org.scalatest.mock.MockitoSugar
+import org.scalatest.{FunSuite, Matchers}
+import slick.driver.JdbcDriver.backend.Database.dynamicSession
+import slick.jdbc.StaticQuery.interpolation
+import slick.jdbc.{StaticQuery => Q}
+
+class VerificationServiceSpec extends FunSuite with Matchers {
+
+  val mockVerificationService = MockitoSugar.mock[VerificationService]
+  val mockEventBus = MockitoSugar.mock[DigiroadEventBus]
+  val mockRoadLinkService = MockitoSugar.mock[RoadLinkService]
+  val verificationDao = new VerificationDao
+
+  lazy val dataSource = {
+    val cfg = new BoneCPConfig(OracleDatabase.loadProperties("/bonecp.properties"))
+    new BoneCPDataSource(cfg)
+  }
+
+  object ServiceWithDao extends VerificationService(mockEventBus, mockRoadLinkService){
+    override def withDynTransaction[T](f: => T): T = f
+    override def dao: VerificationDao = new VerificationDao
+  }
+
+  def runWithRollback(test: => Unit): Unit = TestTransactions.runWithRollback(dataSource)(test)
+
+  test("get verification asset types info"){
+    runWithRollback {
+      val id = sql"""select primary_key_seq.nextval from dual""".as[Long].first
+      sqlu"""insert into municipality_verification (id, municipality_id, asset_type_id, verified_date, verified_by)
+           values ($id, 235, 30, (sysdate - interval '1' year), 'testuser')""".execute
```

Comment:
We can use the "primary_key_seq.nextval" directly on insert query, like for example:

...values (primary_key_seq.nextval,... avoinding a "select" to database.

Reply:
I think it is useful to keep it like this if we ever need to return the ID in the future. This query is at the moment performed max 8 times with the verification service, i don't think its too resource heavy.